FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeProcessingJobSpliter.cs

```diff
@@ -0,0 +1,291 @@
+                    lastSplitTimestamp = DateTimeOffset.Now;
+
+                    yield return new SubJobInfo
+                    {
+                        ResourceType = resourceType,
+                        TimeRange = new TimeRange() { DataStartTime = lastEndTime, DataEndTime = (DateTimeOffset)nextJobEnd },
+                        JobSize = jobSize,
+                    };
+
+                    jobCount += 1;
+                }
+
+                _logger.LogInformation($"Spliting {resourceType} jobs, finish split {jobCount} jobs. use : {(DateTimeOffset.Now - splitingStartTime).TotalMilliseconds} milliseconds.");
+            }
+
+        }
+
+        private async Task<Dictionary<DateTimeOffset, int>> InitializeAnchorListAsync(string resourceType, DateTimeOffset? startTime, DateTimeOffset endTime, int totalCount, CancellationToken cancellationToken)
+        {
+            var anchorList = new Dictionary<DateTimeOffset, int>();
+            if (startTime != null)
+            {
+                anchorList[(DateTimeOffset)startTime] = 0;
+            }
+
+            // Set isDescending parameter false to get first timestmp.
+            var nextTimestamp = await GetNextTimestamp(resourceType, startTime, endTime, false, cancellationToken);
+
+            // Set isDescending parameter as true to get last timestmp.
+            var lastTimestamp = await GetNextTimestamp(resourceType, startTime, endTime, true, cancellationToken);
+
+            if (nextTimestamp != null)
+            {
+                anchorList[(DateTimeOffset)nextTimestamp] = 0;
+            }
+
+            if (lastTimestamp != null)
+            {
+                anchorList[(DateTimeOffset)lastTimestamp] = totalCount;
+            }
+
+            anchorList[endTime] = totalCount;
+
+            return anchorList;
+        }
+
+        // get the lastUpdated timestamp of next resource for next processing job
+        private async Task<DateTimeOffset?> GetNextTimestamp(string resourceType, DateTimeOffset? start, DateTimeOffset end, bool isDescending, CancellationToken cancellationToken)
+        {
+            List<KeyValuePair<string, string>> parameters = new List<KeyValuePair<string, string>>
+            {
+                new KeyValuePair<string, string>(FhirApiConstants.LastUpdatedKey, $"lt{end.ToInstantString()}"),
+                new KeyValuePair<string, string>(FhirApiConstants.PageCountKey, FhirApiPageCount.Single.ToString("d")),
+                new KeyValuePair<string, string>(FhirApiConstants.TypeKey, resourceType),
+            };
+
+            if (isDescending)
+            {
+                parameters.Add(new KeyValuePair<string, string>(FhirApiConstants.SortKey, "-_lastUpdated"));
+            }
+            else
+            {
+                parameters.Add(new KeyValuePair<string, string>(FhirApiConstants.SortKey, "_lastUpdated"));
```

Comment:
Does this sort cause the FHIR server ( cosmos DB ) to have a large delay? 
Can we sort the result after getting bundle response from our side?

Reply:
We can not request all resources one time.